### PR TITLE
Contourf matrices

### PR DESCRIPTION
**Pull request by egeerardyn**
This is an intermediate fix for `contourf` plots in HG2 where the `X` and `Y` arguments are matrices.
This allows for the whole contour plot to have special figures as discussed in #721.

What doesn't work yet (as indicated by a `FIXME` in the code), is showing the boundary of the contour surface. We now show the bounding box instead. The actual contours, however, seem alright.

You can try with the following MWE
```matlab
N = 20;
[X,Y] = meshgrid([1:N],[1:N]);
subplot(1,3,1);
contourf(X,Y,peaks(N),50,'edgecolor','blue','linestyle','--')
subplot(1,3,2);
contourf(X+Y/2,Y,peaks(N),50,'edgecolor','none')
subplot(1,3,3);
contourf(X+sin(Y),Y-cos(X),peaks(N),50,'edgecolor','red')
matlab2tikz('test.tikz','standalone',true)
!pdflatex test.tikz
open test.pdf
```

screen shot 2015-08-11 at 14 12 58

### CODE REVIEW

File: src/matlab2tikz.m [hunk 1]
```diff
@@ -2402,11 +2402,12 @@ function assertRegularAxes(handle)
     % Add zero level fill
     xdata = get(h,'XData');
     ydata = get(h,'YData');
+    %FIXME: determine the contour at the zero level not just its bounding box
```
Comment (PeterPablo) [inline src/matlab2tikz.m]: Add `(see #721)` here?

**Comment by PeterPablo** [comment]
Reviewed code and runtime tested. Adapted the ACID test to have a different color. Everything good :+1: 
Feel free to merge.

**Comment by egeerardyn** [comment]
I haven't adapted the ACID to match this, I just had to change the hash because now the line color and everything is included in the TikZ output as well. I found another (slightly related bug, but that's for another time).

**Comment by PeterPablo** [comment]
I know. With my comment I wanted to say that I did test changing the `EdgeColor` of the contour to verify that it is working.

File: src/matlab2tikz.m (hunk 1 above)
Comment (egeerardyn) [inline src/matlab2tikz.m]: :+1:

**Comment by egeerardyn** [comment]
Thanks for the review and useful comments!

**Pull request merged**